충남대 FE 정지원 - 2단계 선물하기 메인 페이지 UI 구현하기

jjw5655 opened the pull request:
1. Emotion에 Color, Typography 토큰 설정
2. Mobile First Design 기반의 Layout 구현
3. 상단 네비게이션 바 영역 구현
4. 카테고리 섹션 구현
5. 기타 섹션 (선물할 친구 선택, 카테캠 화이팅) UI 구현
6. 실시간 급상승 선물랭킹 구현

jjw5655 committed:
feat: Mock데이터 추가, 구조 설정 및 경로 매핑 추가

jjw5655 committed:
feat: Design Token과 Emotion Theme 시스템 구축

- 색상, 타이포그래피, 간격 토큰 정의
- ThemeProvider 적용 및 TypeScript 타입 정의
- Mobile First Design 적용 (720px)
- 카카오 브랜드 컬러 및 기존 컴포넌트 테마 적용

jjw5655 committed:
feat: Mobile First Layout 시스템 구현

- MobileLayout, Header, Main, Section, Container 컴포넌트 생성
- 720px 제한 모바일 중심 레이아웃 적용
- Sticky Header 및 터치 친화적 인터페이스 구현
- 재사용 가능한 Section/Container 시스템 구축
- App.tsx 새 레이아웃 시스템으로 전환

jjw5655 committed:
feat: 상단 네비게이션 바 및 페이지 시스템 구현

- NavigationBar, BackButton, ProfileButton 컴포넌트 구현
- HomePage와 LoginPage 분리 및 상태 기반 라우팅
- 페이지별 네비게이션 설정 (뒤로가기/프로필 버튼 표시/숨김)
- 카카오 브랜드 로그인 페이지 UI 구현
- Design Token 및 Mobile First 디자인 적용
- 터치 친화적 44px 버튼 영역 및 호버 효과

jjw5655 committed:
refactor: 설정 파일 중복 제거 및 App.tsx 리팩토링

- tsconfig.app.json, vite.config.ts 중복 코드 정리
- App.tsx 페이지 기반 구조로 리팩토링
- 코드 가독성 및 유지보수성 향상

jjw5655 committed:
feat: 카테고리 섹션 5x3 그리드 레이아웃 구현

- CategoryItem, CategoryGrid, CategorySection 컴포넌트 구현
- CSS Grid 기반 5x3 레이아웃
- categories.ts의 15개 데이터 사용.
- 터치 친화적 인터랙션 및 Design Token 활용
- 이미지 URL/이모지 모두 지원하는 아이콘 시스템
- HomePage 통합 및 @/category 경로 매핑 추가

jjw5655 committed:
feat: 기타 섹션 컴포넌트 구현 (친구 선택 + 프로모션 섹션)

- FriendSelectSection: 노란색 버튼 + 친구 선택 텍스트
- PromoSection: 카카오 옐로우 배경 + 카테캠 화이팅 메시지
- misc(기타) 컴포넌트 폴더 구성 및 경로 매핑 추가
- HomePage 완전 리팩토링: 카테고리 + 친구선택 + 프로모션 섹션 통합

jjw5655 committed:
feat: 실시간 급상승 선물랭킹 섹션 구현

- RankingSection: 메인 랭킹 섹션 컨테이너
- FilterButtonGroup: 이중 필터 시스템 (대상/랭킹타입)
  * 상단: 전체/여성이/남성이/청소년이
  * 하단: 받고싶어한/많이선물한/위시로받은
- ProductGrid: 3x2→3x7 확장 가능한 반응형 그리드
- ProductCard: 순위배지(1,2,3등 빨간색 강조) + 상품정보
- MoreButton: 더보기 기능으로 6개→21개 상품 확장
- BBQ 데이터 활용하여 21개 랭킹 상품 생성
- ranking 컴포넌트 폴더 구성 및 경로 매핑 추가
- HomePage에 랭킹 섹션 통합하여 완전한 메인페이지 완성

jjw5655 committed:
docs: README.md 작성

jjw5655 committed:
fix: build오류 해결, 미사용 파라미터 제거

Leejin-Yang commented on src/components/TestComponent.tsx:
이전에 작성한 컴포넌트에 테스트 해보는 꼼꼼함이 보이네요 👍
이제 테스트 컴포넌트는 보내주고 사용하는 컴포넌트만 남기도록 해보는거 어떨까요?

Leejin-Yang commented on src/components/StyledButton.tsx:
요 친구도 사용하지 않는다면 이제 보내주어도 좋아보여요 😊

Leejin-Yang commented on src/styles/colors.ts:
```diff
@@ -0,0 +1,107 @@
+// 색상 토큰 정의
+export const colors = {
```
테마를 역할에 따라 나눠서 잘 작성해주었네요 👍

Leejin-Yang commented on src/styles/types.ts:
```diff
@@ -0,0 +1,21 @@
+    typography: typeof typography;
+    spacing: typeof spacing;
+  }
+} 
```
테마 타입 정의와 확장까지 👍

Leejin-Yang commented on src/components/category/CategoryGrid.tsx:
```diff
@@ -0,0 +1,50 @@
+  }
+`;
+
+export interface CategoryData {
+  themeId: number;
+  name: string;
+  image: string;
+}
```
데이터 관련 타입은 추후 재사용 될 수 있다면 별도로 관리하는 것도 좋아보여요 😊

jjw5655 replied:
현재 /src/components/category/CategoryGrid.tsx에 CategoryData타입이 정의되어 있는데, 타입을 분리할 때 아래 두 방법중 뭐가 더 나을까요?
1. 동일한 디렉토리(/src/components/category/)에 types.ts파일을 만들어서 타입을 분리
2. 별도의 외부 디렉토리(/src/types)에 category.ts파일을 만들어서 타입을 분리

뭔가 관련있는 것들은 하나의 폴더에 다 넣어두는게 좋아보이는데, 리뷰어님의 의견이 궁금합니다.

Leejin-Yang replied:
음.. 저는 상황에 따라 달라질 수 있다고 생각해요.

현재는 컴포넌트 내부에서만 사용하니 동일한 디렉토리에 두어도 충분해 보여요. 다만, 추후에 카테고리 데이터를 외부에서 받아온다거나, 다른 모듈에서 사용하는 상황이 생긴다면 외부 디렉토리에서 관리할 수 있어보여요. 프로젝트를 확장하면서 충분히 리팩터링 할 수 있는 부분이니 꾸준히 고민해보시면 좋겠습니다 👍

Leejin-Yang commented on src/components/category/CategoryGrid.tsx:
```diff
@@ -0,0 +1,50 @@
+  onCategoryClick?: (category: CategoryData) => void;
+}
+
+const CategoryGrid: React.FC<CategoryGridProps> = ({ 
```
FC 타입을 따로 작성하신 이유가 있을까요?

jjw5655 replied:
TS에게 이것이 (함수가 아닌) 리액트 컴포넌트임을 확실히 알려주기위해 : React.FC<>를 쓰는것으로 알고있습니다. 없어도 동일하게 동작하지만, 이걸 넣으면 자동완성이나 에러 체크 등을 TS가 알아서 잘 해준다고 하네요.

찾아보니 요즘은 FC를 안쓰는 추세라고 하고, children과 관련된 문제등으로 쓰지 않는게 좋아보이네요. 모든 컴포넌트에서 FC를 다 삭제했습니다.

Leejin-Yang commented on src/components/layout/Container.tsx:
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+const ContainerWrapper = styled.div<{ fullWidth?: boolean }>`
```
스타일드를 아주 잘 활용하셨네요 👍

한가지 팁을 드리자면 이 타입의 경우 Props에서 받아온 값을 그대로 쓰고 있기 때문에 [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys)을 사용할 수 있어보여요. 이 기회에 유틸리티 타입도 함께 공부해보는걸 추천드려요 😊

jjw5655 replied:
pick과 유틸리티 타입 기억해 두겠습니다!

Leejin-Yang commented on src/components/layout/Container.tsx:
```diff
@@ -0,0 +1,33 @@
+  fullWidth?: boolean;
+}
+
+const Container: React.FC<ContainerProps> = ({ children, fullWidth = false }) => {
```
기본값 설정도 아주 잘 해주셨네요 👍

Leejin-Yang commented on src/components/ranking/ProductCard.tsx:
```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+interface Product {
+  id: string;
+  productId: number;
+  productName: string;
+  price: number;
+  brandName: string;
+  image: string;
+  rank: number;
+  isTopThree: boolean;
+}
```
이 타입도 위의 내용처럼 별도로 관리하는게 좋아보여요!

Leejin-Yang commented on src/components/ranking/RankingSection.tsx:
```diff
@@ -0,0 +1,94 @@
+  { value: 'MANY_WISH', label: '받고 싶어한' },
+  { value: 'MANY_RECEIVE', label: '많이 선물한' },
+  { value: 'MANY_WISH_RECEIVE', label: '위시로 받은' }
+];
```
옵션들은 상수로 관리하면서 위의 타입을 추출하는 방법을 시도해보는 걸 추천드려요. 옵션이 수정되는 경우, 관리하기 좀 더 쉬워진다는 장점이 있습니다. 이런 시도들을 한다면 타입스크립트를 좀 더 잘 다룰 수 있을거에요 😊

Leejin-Yang commented on src/components/ranking/ProductCard.tsx:
```diff
@@ -0,0 +1,104 @@
+        </RankBadge>
+        <ProductImage 
+          src={product.image} 
+          alt={product.productName}
+          loading="lazy"
```
img 태그의 alt 속성과 lazy loading까지 적용해주셨네요 👍

Leejin-Yang commented on src/components/ranking/RankingSection.tsx:
```diff
@@ -0,0 +1,94 @@
+  const rankingProducts = generateRankingProducts();
+
+  const handleTargetChange = (value: string) => {
+    setTargetType(value as TargetType);
```
타입 단언을 사용하지 않기 위해서는 어떤 방법이 있을까요?

jjw5655 replied:
타입 단언에 대해 찾아보았습니다. 타입 단언이란 TS에게 "이 타입이 맞으니 타입검사 하지말고 넘어가"도록 하는 것이며, 타입 안정성이나 런타임에러 등의 문제점이 있다고 합니다. 이런 문제점이 있지만 실제로는 생산성을 위해 많이 사용된다고 하네요.

해결방안은
1. 제네릭 함수로 해결하거나,
2. 옵션 배열에서 직접 찾거나,
3. 타입가드 함수를 사용하거나,
4. 조건문으로 검증하고 사용하는 등의 방법이 있다고 합니다.

Leejin-Yang commented on src/styles/index.ts:
```diff
@@ -0,0 +1,5 @@
+export { default as theme } from './theme';
+export { colors, semanticColors } from './colors';
+export { typography } from './typography';
+export { spacing } from './spacing';
```
theme에서 이 값들은 포함된 걸로 보이는데 따로 export한 이유가 있을까요?

jjw5655 replied:
일단 폴더 내에 모든걸 다 내보낸다는 통일성때문에 넣은것이기도 하고, 각각이 개별적으로 export되서 사용될 수도 있지 않을까?라는 생각때문이기도 합니다.
이 styles 폴더의 경우 Theme에서 어차피 다 모아다가 export하니까 없어도 될 것 같습니다.

Leejin-Yang commented on src/components/category/CategoryGrid.tsx:
저는 컴포넌트 가독성을 위해 스타일 관련 코드들은 컴포넌트 아래로 작성하곤 해요. 또는 스타일 관련은 파일을 분리하곤 합니다.
이 부분 참고해서 지원님에게는 어떤 방식이 더 좋을지 고민해보면 좋을 것 같아요 😊

jjw5655 replied:
음.. 제 생각은, HTML구조(TSX)가 딱 박혀있는 컴포넌트가 코드 가장 하단부에 있는 코드구조를 가장 많이봐서 이게 가장 익숙하긴 합니다. 그래서 저는 그냥 이 상태 그대로 두거나, 차라리 스타일만 따로 분리하는게 더 좋아보여요.

스타일 분리하면 또 파일이 추가가 돼서, 지금 이 코드의 경우 그냥 그대로 둘 것 같습니다.

Leejin-Yang commented on src/components/layout/Header.tsx:
```diff
@@ -0,0 +1,47 @@
+  );
+};
+
+export default Header; 
```
컴포넌트 파일들이 eol이 없는 것으로 보이네요
Prettier 적용이 잘 되는지, formatOnSave 옵션이 설정 되어 있는지 확인해주세요 😊

참고로 PR 작성할 때, 제출 전에 확인할 수 있어요

jjw5655 replied:
프리티어 모두 적용했습니다. PR 제출 전에 확인하는것도 해보겠습니다!

자세한 리뷰 감사드립니다! 제가 리액트와 코딩 실력이 많이 부족하여 상당부분을 AI에 의존하고 있어요. AI가 짜준 코드를 전부 이해하진 못해서 전체적인 코드 구조를 잡아보거나, "너가 짜준 코드중에 중요한 부분들을 알려달라"고 요청해서 공부하고 넘어가는 식인데요. 리뷰어님이 잘 구현된 부분이나 중요한 부분, 부족한 부분들을 많이 집어주셔서 이 부분을 자세히 알아보면서 공부가 많이 되고있습니다. 감사드립니다.

Leejin-Yang replied:
저는 ai를 활용할 때 좀 더 깊게 이해하려고 노력하는 편이에요.
어떤 의도로 작성했는지 설명해야하는 상황은 항상 오기 때문에, ai가 제안한 코드를 스스로 분석해보려고 해요.
그리고 더 좋은 방법이 있나 다시 시도해보기도 하구요.

지원님은 지금도 충분히 잘 하고 있다고 생각해요. 리뷰를 하면서 고민하신 흔적들이 보이고, 앞으로도 미션을 진행하면서 이런 고민들을 계속 이어가신다면 더 빠르게 성장할 수 있을거에요 👍 저도 최대한 도움을 드리도록 노력하겠습니다. 궁금한 점이 있으면 언제든 DM 주세요 😊

Leejin-Yang reviewed (changes requested):
지원님 안녕하세요!
프론트엔드 개발의 시작점에 오셨네요 🙌 UI 구현하느라 고생하셨어요!

전반적으로 매우 잘 구현해주셔서 놀랐습니다 👍 리뷰 외에 몇가지 더 이야기를 드리자면,
미션 진행하면서 컴포넌트를 작성할 때 이런 부분들을 더 고민해보면 좋아보여요

- 이 컴포넌트가 어떤 역할까지 책임져야 하는지
- 내부에 포함해야 할 로직과 외부 prop으로 받아야 할 값을 어떤 기준으로 정해야 할지
- 어떤 컴포넌트를 재사용 할 수 있을지

처음 작성할 때부터 하면 어려우니 미션 진행하면서 아마 이런 고민을 하는 순간이 올거에요 😊
(저도 늘 고민하는 부분이긴 합니다 😅)

그리고 실행하면서 확인한건데, 아래 이미지처럼 모바일 화면에서 그리드가 깨져보이네요. 이 부분도 함께 수정해주세요 😊 추가로 선물랭킹에서 접기 기능 구현까지 도전해보세요 👍

<img width="374" alt="Screenshot 2025-06-27 at 08 13 41" src="https://github.com/user-attachments/assets/95e76fbe-de31-41fe-bb70-2699604b7c70" />

리뷰 확인하시고 다시 리뷰 요청 보내주세요 😊
그리고 궁금한 점은 언제든지 코멘트로 남겨주세요

jjw5655 committed:
remove: 미사용 컴포넌트 파일 제거 (StyledButton, TestComponent)

jjw5655 committed:
refactor: 전체 프로젝트에서 React.FC 제거

- 모든 컴포넌트에서 React.FC를 제거하고 명시적 props 타입 사용
- 더 간결하고 명확한 타입 정의로 변경
- 최신 React TypeScript 패턴 적용

jjw5655 committed:
refactor: 상수에서 타입 추출하여 타입 안전성 개선

- RankingSection: 타입을 상수에서 추출하도록 변경
- FilterButtonGroup: readonly 배열 지원 추가
- 옵션 수정 시 한 곳만 변경하면 되도록 개선

jjw5655 committed:
fix: prettierrc 주석 제거 및 default값 정리

jjw5655 committed:
style: prettier 포맷팅 적용

jjw5655 committed:
fix: 모바일에서 CategoryGrid  화면 넘침 문제 해결

jjw5655 committed:
feat: RankingSection에 접기 기능 추가

jjw5655 commented:
모바일 화면에서 그리드가 넘어가는 문제와, 선물랭킹에서 접기 기능까지 구현했습니다. 👍

Leejin-Yang commented on .prettierrc:
```diff
@@ -1,8 +1,4 @@
-  "useTabs": false // 스페이스 사용하여 들여쓰기 설정 (default: false)
-} 
\ No newline at end of file
+  "trailingComma": "es5",
+  "singleQuote": true
+}
```
👍

Leejin-Yang commented on src/App.tsx:
```diff
@@ -1,52 +1,73 @@
 function App() {
+  const [currentPage, setCurrentPage] = useState<PageType>('home');
+
+  // 페이지별 네비게이션 설정
+  const getNavigationConfig = () => {
```
곳곳에 주석들이 많이 보이는데요, 코드로 충분히 설명이 가능하다면 따로 주석을 남기지 않는 것이 좋아보여요 😊

Leejin-Yang commented on src/components/ranking/RankingSection.tsx:
```diff
@@ -0,0 +1,91 @@
+
+  const handleRankChange = (value: string) => {
+    setRankType(value as RankType);
+    // URL 파라미터 업데이트는 추후 라우팅 시 구현
```
주석과 이어지는 내용으로, 소소한 이야기지만
추후 업데이트가 필요한 부분은 작성해주신 것처럼 TODO로 남겨두기도 해요. 다만 다른 주석과 구분할 수 있게 TODO를 prefix로 붙이는 방법도 있어요

Leejin-Yang commented on src/components/category/CategoryGrid.tsx:
```diff
@@ -0,0 +1,52 @@
+  gap: ${(props) => props.theme.spacing.spacing2};
+  width: 100%;
+
+  /* 모바일에서 4열로 변경하여 화면 넘침 방지 */
+  @media (max-width: 480px) {
+    grid-template-columns: repeat(4, 1fr);
+    gap: ${(props) => props.theme.spacing.spacing1};
+  }
+
+  /* 더 작은 화면에서는 3열로 */
+  @media (max-width: 360px) {
+    grid-template-columns: repeat(3, 1fr);
+  }
```
👍

Leejin-Yang reviewed (approved):
지원님 안녕하세요! 리뷰를 꼼꼼하게 봐주셔서 감사합니다 ☺️

요구사항도 만족하고, 실행했을 때 특별히 이슈가 없이 이번 단계는 여기서 merge 할게요 👍
몇개 소소한 코멘트 남겼는데 참고해주세요.!

다음 단계에서 보아요, 고생 많으셨습니다 🙌
